src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -4445,7 +4444,18 @@ FindReferencedTableColumn(Expr *columnExpression, List *parentQueryList, Query *
 	 */
 	if (candidateColumn->varlevelsup > 0)
 	{
-		return;
+		int parentQueryIndex = list_length(parentQueryList) -
+							   candidateColumn->varlevelsup;
+		if (parentQueryIndex < 0)
+		{
+			return;
+		}
+
+		/*
+		 * as we get the query from varlevelsup up, we reset the varlevelsup.
+		 */
+		candidateColumn->varlevelsup = 0;
+		query = list_nth(parentQueryList, parentQueryIndex);
```

Comment:
Should we be also adjusting the `parentQueryList`? I guess it might be necessary because we still need the accurate parentQueryList for to hold for `ctelevelsupctelevelsup`.

Could you find a query where we both have `ctelevelsupctelevelsup > 0` and `varlevelsup > 0`?

Reply:
> Could you find a query where we both have ctelevelsupctelevelsup > 0 and varlevelsup > 0?

Is this possible with the current patch?

Reply:
Even if not, I think we should add a comment here warning the future code changes to be careful about it.

Reply:
I have added this but I guess it doesn't have ctelevelsup > 0.

https://github.com/citusdata/citus/pull/5079/files#diff-fb756d0d31008052220149d21e9990dccba97f3b65e643d860d271ff565ed1c9R1159

Anyways, I will add a comment either way